crates/types/src/config/object_store.rs

```diff
@@ -18,35 +18,41 @@ use serde_with::serde_as;
 #[serde(rename_all = "kebab-case")]
 #[builder(default)]
 pub struct ObjectStoreOptions {
-    /// The AWS configuration profile to use for S3 object store destinations.
+    /// The AWS configuration profile to use for S3 object store destinations. If you use
+    /// named profiles in your AWS configuration, you can replace all the other settings with
+    /// a single profile reference. See the [AWS documentation on profiles]
+    /// (https://docs.aws.amazon.com/sdkref/latest/guide/file-format.html) for more.
     #[serde(skip_serializing_if = "Option::is_none")]
     pub aws_profile: Option<String>,
 
-    /// AWS region to use with S3 object store destinations.
+    /// AWS region to use with S3 object store destinations. This may be inferred from the
+    /// environment, for example the current region when running in EC2. Because of the
+    /// request signing algorightm this must have a value. For Minio, you can generally
+    /// set this to any string, such as `us-east-1`.
```

Comment:
Does this need to be set for minio, if not, can we say that it has no effect on other object stores?

Reply:
It must be set for the S3 API regardless of the server implementation, even for Minio, as the comment says. The SigV4 algorithm that requires a region as one of the inputs. I thought about defaulting it to _something_ but that introduces a risk of other problems for AWS customers which would be harder to troubleshoot than just failing with "missing region".

Reply:
I've not set this in my test and it worked fine. Is request signing associated with being on https? (I'd assume no)

Reply:
Is it possible that you have a default region configured, either via `AWS_REGION` env var or in ~/.aws/config perhaps? We will try to source it from somewhere, but if we can't, and it's unset, the S3 client will fail to initialize.

Reply:
And no, AWS request signing is independent of HTTP/S - it's passed as a set of request headers:

https://docs.aws.amazon.com/AmazonS3/latest/API/sigv4-auth-using-authorization-header.html

Reply:
Probably this is what's been happening.